src/lib/gundb/UserStorageClass.js

```diff
@@ -1333,6 +1333,24 @@ export class UserStorage {
     return this.loadGunField(profileNode)
   }
 
+  /**
+   *
+   * @param cb function
+   * @returns {Promise<void>}
+   */
+  onProfile(cb = () => {}) {
+    this.profile.on(encProfile => {
+      if (!encProfile) {
+        return
+      }
+      const publicProfile = this.getDisplayProfile(encProfile)
+      if (!publicProfile.email) {
```

Comment:
this doesn't seem right

Reply:
while gun db saving. I receive empty object with subscribe. The email is one of required field. so I think the best solution is this one.

Reply:
I test this in two browsers. when I change profile in one browser - it changes in another.